climada/engine/impact.py

```diff
@@ -390,6 +392,24 @@ def impact_per_year(self, all_years=True, year_range=None):
             year_set[year] = sum(self.at_event[orig_year == year])
         return year_set
 
+    def impact_per_exp_reg(self, exposures):
```

Comment:
Please add a unit test for this method, using a self-made impact matrix and exposures, and checking the exact result.

Reply:
As the PR proposes a change of the type `Impact`, all functions and methods with an `Impact` in their signature are possibly affected. A single unittest won't do, I'm afraid. In particular, `unsequa`, `cost_benefit` and read/write methods should probably be tested for consistency.

Reply:
thanks for the clarification, if this is the case then this PR does indeed require more care... however, I still do not fully understand how can these classes (e.g., `unsequa`, `cost_benefit`) be affected if none of the methods therein use`Impact.impact_per_exp_reg`.

Reply:
I think @emanuel-schmid refers to the change of the `Impact` signature. If we add a new attribute that can be set via the `Impact.__init__`, a lot of tests would need to be checked and updated. However, already from a design perspective, I think we should not do that (see my https://github.com/CLIMADA-project/climada_python/pull/642#discussion_r1107390523). If we just add a method that returns a value, but otherwise does not modify the attributes of `Impact`, we only need to test that method (and make sure that all other tests pass unchanged)

Reply:
cool, then I will 1) just return a value and 2) make a unit test. Then we should be fine I guess.

Reply:
done